internal/pkg/deploy/deploy.go

```diff
@@ -163,17 +172,29 @@ func (s *Store) IsDeployed(appName string, envName string, svcName string) (bool
 	return false, nil
 }
 
+// // getServiceName gets the ECS service name given a specific ARN.
+// // For example: arn:aws:ecs:us-west-2:123456789012:service/my-app-test-Cluster-cdgG2k6XIBtM/my-app-test-my-svc-Service-WLYA7MGACV1F
+// // returns my-service
+// func (s *Store) getServiceName(svcARN string) (string, error) {
+// 	tags, err := s.ecsClient.GetTags(svcARN)
+// 	if err != nil {
+// 		return "", fmt.Errorf("get tags for ECS service: %w", err)
+// 	}
+// 	if _, ok := tags[ServiceTagKey]; !ok {
+// 		return "", fmt.Errorf("service with ARN %s is not tagged with %s", svcARN, ServiceTagKey)
+// 	}
+// 	return tags[ServiceTagKey], nil
+// }
+
 // getServiceName gets the ECS service name given a specific ARN.
-// For example: arn:aws:ecs:us-west-2:123456789012:service/my-http-service
-// returns my-http-service
-func (s *Store) getServiceName(svcARN string) (string, error) {
-	resp, err := arn.Parse(svcARN)
-	if err != nil {
-		return "", fmt.Errorf("parse service ARN %s: %w", svcARN, err)
-	}
-	resource := strings.Split(resp.Resource, "/")
-	if len(resource) != 2 {
-		return "", fmt.Errorf(`cannot parse service ARN resource "%s"`, resp.Resource)
+// For example: arn:aws:ecs:us-west-2:123456789012:service/my-app-test-Cluster-cdgG2k6XIBtM/my-app-test-my-svc-Service-WLYA7MGACV1F
+// returns my-service
+func (s *Store) getServiceName(svcARN, appName, envName string) (string, error) {
```

Comment:
Using regex could potentially be cheaper than calling ECS API to get service name directly from the service ARN. However, it could be not that safe to do so because it is regex.